gmso/atomtyping/parameterize.py

```diff
@@ -0,0 +1,275 @@
+        # populate molecule information
+        isomorphic_top_graphs = partition_isomorphic_topology_graphs(top_graph)
+        apply_atomtypes_from_isomorphic(
+            top, isomorphic_top_graphs, forcefields
+        )  # what if only one forcefield?, traversing done in this function
+    elif not use_residue_info:  # False, False
+        # get entire topgraph
+        topgraph = get_modified_topology_graph(top)
+        # get rules provider from gmso_ff
+        # Assumes forcefields is just one forcefield
+        at_rules_provider = get_atomtyping_rules_provider(gmso_ff=forcefields)
+        # create a typemap of that topgraph
+        typemap = find_atomtypes(topgraph, at_rules_provider)
+        # apply typemap
+        traverse_typemap(top, typemap, forcefields)
+    elif use_residue_map:  # False, True
+        """Not Implemented
+        maps = {}
+        for key, val in forcefields.items():
+            # generate topgraph of items with "key" residue name
+            # NOTE CAL: this iter_sites_by method won't work currently, can only iter one at a time.
+            res_iter = top.iter_sites_by(["residue_name", "residue_number"], [key, 1]) #Need to note that resnumber index 1
+            # create a top/graph from this set of sites
+            #subgraph = FUNCTION(res_iter)
+            # append that typemap to a dict with residue name as key
+            #maps[key] = find_atomtypes(subgraph, val)
+        # generate a full typemap by iterating through all sites and copying over map info from `maps`
+        #typemap = FUNCTION(top, maps)
+        # apply typemap
+        traverse_typemap(top, typemap, forcefields)
+        """
+        raise (
+            GMSOError,
+            "Using site residue matching to match substructures to a given forcefield is not implemented yet",
+        )
+
+    if assert_bond_params:
+        apply_connection_types(top, forcefields, "bond")
+    if assert_angle_params:
+        apply_connection_types(top, forcefields, "angle")
+    if assert_dihedral_params:
+        apply_connection_types(top, forcefields, "dihedral")
+    if assert_improper_params:
+        apply_connection_types(top, forcefields, "improper")
+
+    return top
+
+
+def get_modified_topology_graph(gmso_topology):
+    """Return a TopologyGraph with relevant attributes from an GMSO topology.
+
+    Parameters
+    ----------
+    gmso_topology: gmso.Topology
+        The GMSO Topology
+
+    Returns
+    -------
+    TopologyGraph
+        The equivalent TopologyGraph of the openFF Topology `openff_topology`
+    """
+    top_graph = TopologyGraph()
+    for atom in gmso_topology.sites:
+        if isinstance(atom, Atom):
+            if atom.name.startswith("_"):
+                top_graph.add_atom(
+                    name=atom.name,
+                    index=gmso_topology.get_index(atom),
+                    atomic_number=None,
+                    element=atom.name,
+                    subtopology_label=atom.label,
```

Comment:
Not sure where we want to address this but we need a better identifier than `atom.label`.